mgplvm/models/bfa.py

```diff
@@ -38,13 +40,25 @@ class Bfa(Module):
 
     def __init__(self,
                  n: int,
+                 d: int,
                  sigma: Optional[Tensor] = None,
-                 learn_sigma=True):
+                 learn_sigma=True,
+                 Y=None):
+
         super().__init__()
         if sigma is None:
-            sigma = torch.ones(n,)  # TODO: FA init
+            if Y is None:
+                sigma = torch.ones(n,)  # TODO: FA init
+            else:
+                n_samples, n, m = Y.shape
+                mod = decomposition.FactorAnalysis(n_components=d)
+                Y = Y.transpose(0, 2, 1).reshape(n_samples * m, n)
+                mudata = mod.fit_transform(Y)  #m*n_samples x d
+                sigma = torch.tensor(np.sqrt(mod.noise_variance_))
+
         self._sigma = nn.Parameter(data=sigma, requires_grad=learn_sigma)
         self.n = n
+        self.likelihood, self.z, self.kernel = [NoneClass() for i in range(3)]
```

Comment:
This doesn't seem like the most elegant solution...
Is the issue here that we need these functions for optimization? If that's the case we should really deal with that in the optimization modules?

Reply:
But fine, maybe we can fix this in a future PR. But perhaps we should add a "TODO" comment here?